bundles/org.eclipse.mita.program/src/org/eclipse/mita/program/scoping/ProgramDslScopeProvider.xtend

```diff
@@ -312,8 +316,7 @@ class ProgramDslScopeProvider extends AbstractProgramDslScopeProvider {
 		if (subType.name == superTypeName) {
 			return true
 		}
-		val match = subType.superTypes.findFirst[name == superTypeName]
-		return match !== null
+		return typeSystem.getSuperTypes(subType).exists[name == superTypeName]
```

Comment:
Remark: Type#getSuperTypes() only returns the direct super types. It does not return the complete super type hierarchy. While ITypeSystem#getSuperTypes() goes the super type hierarchy up and collects all super types.